### PR TITLE
feat: create custom srcsets

### PR DESCRIPTION
**Pull request by ericdeansanchez**
The purpose of this PR is to refactor and normalize the custom
srcset creation interface.

### CODE REVIEW

File: tests/Imgix/Tests/UrlBuilderTest.php [hunk 1]
```diff
@@ -289,7 +297,20 @@ public function testDprSrcsetWithQ100() {
...
+https://demos.imgix.net/image.jpg?dpr=2&ixlib=php-3.2.0&q=50&w=640 2x,
+https://demos.imgix.net/image.jpg?dpr=3&ixlib=php-3.2.0&q=35&w=640 3x,
+https://demos.imgix.net/image.jpg?dpr=4&ixlib=php-3.2.0&q=23&w=640 4x,
+https://demos.imgix.net/image.jpg?dpr=5&ixlib=php-3.2.0&q=20&w=640 5x';
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testDprSrcsetQoverridesVariableQuality() {
+        $builder = new UrlBuilder("demos.imgix.net", true, false);
+        $actual = $builder->createSrcSet($path="image.jpg", $params=array("w" => 640));
```
Comment (sherwinski) [inline tests/Imgix/Tests/UrlBuilderTest.php:313]: Did you mean to pass in an overriding `q` parameter here?

File: tests/Imgix/Tests/UrlBuilderTest.php [hunk 2]
```diff
@@ -302,10 +323,11 @@ public function testDprSrcsetWithDefaultQuality() {
 
     public function testDprSrcsetWithDisabledVariableQuality() {
```
Comment (sherwinski) [inline tests/Imgix/Tests/UrlBuilderTest.php:324]: Maybe just a bad case of copy+paste, but I think we want to test that this srcset produces no `q` parameter when `$disableVariableQuality=true`.

File: tests/Imgix/Tests/UrlBuilderTest.php (hunk 1 above)
Comment (ericdeansanchez) [reply tests/Imgix/Tests/UrlBuilderTest.php]: Yes! Thank you!

File: tests/Imgix/Tests/UrlBuilderTest.php (hunk 2 above)
Comment (ericdeansanchez) [reply tests/Imgix/Tests/UrlBuilderTest.php]: Yeah, there was a couple things up here. I revamped some things and briefly commented three tests that might cause confusion. I also found that I had not completely removed a function parameter (a hold over from removing custom qualities).

**Pull request merged**